src/wallet/rpcwallet.cpp

```diff
@@ -3905,11 +4012,21 @@ UniValue z_viewtransaction(const UniValue& params, bool fHelp)
         }
         auto memo = notePt.memo();
 
+        // If the note belongs to a Sapling address that is part of an account in the
+        // wallet, show the corresponding Unified Address.
+        std::string address;
+        const auto ua = pwalletMain->GetUnifiedForReceiver(pa);
+        if (ua.has_value()) {
+            address = keyIO.EncodePaymentAddress(ua.value());
+        } else {
+            address = keyIO.EncodePaymentAddress(pa);
+        }
```

Comment:
It's ugly that this is duplicated. Maybe use a closure?

Reply:
I tried a few different options, and this seemed like the cleanest.